### PR TITLE
feature: supported to use host name in upstream

### PR DESCRIPTION
**Pull request by membphis**
will fix: https://github.com/iresty/apisix/issues/21

- [x] doc

### CODE REVIEW

File: lua/apisix.lua [hunk 1]
```diff
@@ -143,6 +151,70 @@ function _M.http_ssl_phase()
...
+    local local_conf = core.config.local_conf()
+    local dns_resolver = local_conf and local_conf.apisix and
+                         local_conf.apisix.dns_resolver
+    local new_nodes = core.table.new(0, 8)
+
+    for addr, weight in pairs(up.value.nodes) do
+        local host, port = core.utils.parse_addr(addr)
+        if not ipmatcher.parse_ipv4(host) and
+           not ipmatcher.parse_ipv6(host) then
+            local ip_info = core.utils.dns_parse(dns_resolver, host)
```
Comment (moonming) [inline lua/apisix.lua]: we should not drop the error msg of `dns_parse`. if the domain is invalid or timeout, what's the result? we should add test cases for that.

**Comment by moonming** [comment]
@membphis please take a look comments.

File: conf/config.yaml [hunk 2]
```diff
@@ -29,6 +29,10 @@ apisix:
   #   udp:                        # UDP proxy port list
   #     - 9200
   #     - 9211
+  dns_resolver:                   # use 114.114.114.114 as default DNS resolver, disable IPv6 and enable local DNS.
+    - 114.114.114.114
```
[the remark this answers is not in the document]
Comment (membphis) [reply conf/config.yaml]: After the international lover feel that it is wrong, we modify it better.

**Comment by membphis** [comment]
> @membphis please take a look comments.

you'd better create new issue if you leave new message when we merged some PR already.

**Comment by moonming** [comment]
no, I commented before merged.

YuanSheng Wang <notifications@github.com> 于2019年9月23日周一 下午8:54写道：

**Comment by membphis** [comment]
image

**Pull request merged**